apps/discord-bot/src/utils/analytics.ts

```diff
@@ -114,6 +115,12 @@ export type QuestionSolvedProps = QuestionAskedProps &
 type MarkSolutionUsedProps = UserProps<'User'> & {
 	Status: MarkSolutionErrorReason | 'Success';
 };
+type FeedbackGivenProps = ServerPropsWithDiscordData &
+		ChannelPropsWithDiscordData & 
+		UserProps<'advisor'> & {
+	FeedbackType: 'Bug' | 'Suggestion';
+	CanFollowUp: boolean;
+};
```

Comment:
I want to know why we need source property in this ?

Reply:
> why we need source property 

In the future, we'll be using this feedback event for multiple locations so the source will distinguish between them

>  also are serverProps and ChannelProps needed in this type

Sure lets include them, it'll be helpful to know where the feedback is coming from

> feedbackGiver

Sure, that works for it